packages/remote-feature-flag-controller/src/remote-feature-flag-controller.test.ts

```diff
@@ -0,0 +1,241 @@
+    state: options.state ?? {},
+    clientConfigApiService:
+      options.clientConfigApiService ?? buildClientConfigApiService(),
+    disabled: options.disabled,
+  });
+}
+
+describe('RemoteFeatureFlagController', () => {
+  let clientConfigApiService: AbstractClientConfigApiService;
+
+  beforeEach(() => {
+    clientConfigApiService = buildClientConfigApiService();
+    jest.clearAllMocks();
+  });
+
+  describe('constructor', () => {
+    it('should initialize with default state', () => {
+      const controller = createController();
+
+      expect(controller.state).toStrictEqual({
+        remoteFeatureFlags: {},
+        cacheTimestamp: 0,
+      });
+    });
+
+    it('should initialize with disabled parameter', () => {
+      const controller = createController({ disabled: true });
+
+      expect(controller.state).toStrictEqual({
+        remoteFeatureFlags: {},
+        cacheTimestamp: 0,
+      });
+    });
+  });
+
+  describe('getRemoteFeatureFlags', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should get feature flags from API when cache is invalid', async () => {
+      const controller = createController({ clientConfigApiService });
+
+      const flags = await controller.getRemoteFeatureFlags();
+
+      expect(flags).toStrictEqual(mockFlags);
+    });
+
+    it('should return empty object when disabled', async () => {
+      const controller = createController({
+        clientConfigApiService,
+        disabled: true,
+      });
+
+      const remoteFeatureFlags = await controller.getRemoteFeatureFlags();
+
+      expect(remoteFeatureFlags).toStrictEqual({});
+      expect(clientConfigApiService.fetchFlags).not.toHaveBeenCalled();
+    });
+
+    it('should properly enable controller and make network request', async () => {
+      const controller = createController({
+        clientConfigApiService,
+        disabled: true,
+      });
+
+      controller.enable();
+
+      const remoteFeatureFlags = await controller.getRemoteFeatureFlags();
+      expect(remoteFeatureFlags).toStrictEqual(mockFlags);
+      expect(clientConfigApiService.fetchFlags).toHaveBeenCalledTimes(1);
+    });
+
+    it('should properly disable controller and not make network request', async () => {
+      const controller = createController({
+        clientConfigApiService,
+        disabled: false,
+      });
+
+      controller.disable();
+
+      const remoteFeatureFlags = await controller.getRemoteFeatureFlags();
+      expect(remoteFeatureFlags).toStrictEqual({});
+      expect(clientConfigApiService.fetchFlags).not.toHaveBeenCalled();
+    });
+
+    it('should not affect existing cache when toggling disabled state', async () => {
+      const controller = createController({
+        clientConfigApiService,
+        disabled: false,
+      });
+
+      // First, enable and get flags to populate cache
+      await controller.getRemoteFeatureFlags();
```

Comment:
Nit: Similar deal as the `controller.disable()` line above. If you know that `getRemoteFeatureFlags` changes the state, then when you instantiate the controller would it makes sense to place it in the right state ahead of time? This includes disabling the controller as well.

Reply:
Good catch! I agree—it makes sense to initialize the controller in the expected state upfront, rather than toggling it during the test. It makes the tests clearer and avoids unnecessary steps like calling `disable()` manually. I'll update the related tests to set the disabled state directly when creating the controller. Thanks for pointing this out!